synapse/lib/types.py

```diff
@@ -507,7 +507,7 @@ def __init__(self):
         self.addSubType('inet:passwd','str')
         self.addSubType('inet:filepath','str')
 
-        self.addSubType('inet:fqdn','str', regex='^[a-z0-9.-_]+$', lower=1)
+        self.addSubType('inet:fqdn','str', regex='^[a-z0-9.-_\-]+$', lower=1)
```

Comment:
I think it may be reading `.-_` as a range? Maybe just escape that dash instead of adding another?